client/src/components/Navbar/Navbar.tsx

```diff
@@ -110,18 +115,36 @@ const Navbar: React.FC = () => {
     handleClose();
     logout();
   };
+  const [profile, setProfile] = useState<Profile>();
 
   const renderMenuItems = () => {
-    // TODO: conditionally render based on profile type
+    // Add a different view if the profile is null? Have a link suggesting them to create a profile?
     return menuItems.map((menu) => {
-      if (menu.authenticated) {
+      if (menu.authenticated && menu.canView?.includes(profile?.accountType || '')) {
         return loggedInUser && <MenuItem key={menu.resource} {...menu} />;
       } else {
-        return !loggedInUser && <MenuItem key={menu.resource} {...menu} />;
+        if (!menu.authenticated) {
+          return !loggedInUser && <MenuItem key={menu.resource} {...menu} />;
+        }
       }
     });
   };
 
+  useEffect(() => {
+    if (loggedInUser) {
```

Comment:
Good work! Eventually we can store the logged in user's profile within a context (maybe the auth context) - so that it is accessible throughout the app